src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -216,36 +210,45 @@ internal static object GetRuntimeClassCCWTypeIfAny(object obj)
             else
             {
                 var objType = type.GetRuntimeClassCCWType() ?? type;
-                var interfaces = objType.GetInterfaces();
-                foreach (var iface in interfaces)
+                // Check whether the type itself has the attribute to make sure it is using the new source generator
+                // and just doesn't have an updated dependent projection which is.
+                if (type.GetCustomAttribute(typeof(WinRTExposedTypeAttribute), false) is IWinRTExposedTypeDetails winrtExposedClassAttribute)

```

Comment:
Just so I understand, why the `IWinRTExposedTypeDetails` instead of just accessing the attribute directly? Wouldn't this also add a tiny bit more overhead due to the interface calls and the extra interface on the attribute?

Reply:
I may end up removing the interface, but the interface is there atm as for both generic interface scenarios and possibly for structs and delegates, I was thinking about generating unique attributes extending `WinRTExposedTypeAttribute` that have custom constructors that provide more flexibility to get types and so on while knowing the type information without reflection.  I was debating between doing that vs structs and delegates getting their own common unique attribute type.  I most likely will just go with extending instead so I probably would be able to get rid of the interface.